In [ ]:
import pandas as pd

csv_path = "../data/datatrain.csv"  

df = pd.read_csv(csv_path)

print("Kolom CSV:", df.columns.tolist())

print("\nUnique values di kolom emotion:")
print(df["emotion"].unique())

print("\nValue counts:")
print(df["emotion"].value_counts())


Kolom CSV: ['id', 'video', 'emotion']

Unique values di kolom emotion:
['Surprise' 'Proud' 'Trust' 'Sadness' 'Terkejut' 'Joy' 'Anger' 'Trkejut'
 'Fear' 'Neutral' 'Love' 'Marah' 'Kaget' 'Loyalty' 'Terkjut' 'Faith'
 'Percaya' 'Bangga' 'Sad' 'terkejut' 'kaget' 'Marh' 'Trekejut']

Value counts:
emotion
Surprise    320
Trust       181
Proud       154
Joy          53
Anger        34
Sadness      17
Fear         16
Neutral       8
Terkjut       3
Terkejut      2
Trkejut       2
Bangga        2
Love          1
Kaget         1
Marah         1
Loyalty       1
Faith         1
Percaya       1
Sad           1
terkejut      1
kaget         1
Marh          1
Trekejut      1
Name: count, dtype: int64


In [2]:
import pandas as pd
import os

# Path dan output file
input_path = '../data/raw/datatrain.csv'
output_dir = '../data/processed'
output_path = os.path.join(output_dir, 'train_clean.csv')

# Load the CSV file
df_train = pd.read_csv(input_path)

# Dictionary untuk mengoreksi dan menyamakan nilai emosi
emotion_map = {
    'Terkejut': 'Surprise',
    'Trkejut': 'Surprise',
    'Terkjut': 'Surprise',
    'Trekejut': 'Surprise',
    'Kaget': 'Surprise',
    'kaget': 'Surprise',
    'terkejut': 'Surprise',
    'Marah': 'Anger',
    'Marh': 'Anger',
    'Bangga': 'Proud',
    'Percaya': 'Trust',
    'Loyalty': 'Trust',
    'Faith': 'Trust',
    'Sad': 'Sadness'
}

# mengganti nilai yang tidak konsisten di kolom 'emotion'
df_train['emotion'] = df_train['emotion'].replace(emotion_map)

# menghapus baris yang memiliki emosi 'Love' karena tidak termasuk dalam 8 kategori BDC
df_train = df_train[df_train['emotion'] != 'Love']

# memastikan folder output ada, jika tidak, buat
os.makedirs(output_dir, exist_ok=True)

# menyimpan DataFrame yang sudah dibersihkan ke file CSV
df_train.to_csv(output_path, index=False)

print(f"\nDataFrame yang sudah dibersihkan sudah disimpan di '{output_path}'")
print(f"Sekarang total baris: {len(df_train)} (satu baris 'Love' telah dihapus).")
print("\nNilai unik setelah perbaikan:")
print(df_train['emotion'].unique())


DataFrame yang sudah dibersihkan sudah disimpan di '../data/processed\train_clean.csv'
Sekarang total baris: 802 (satu baris 'Love' telah dihapus).

Nilai unik setelah perbaikan:
['Surprise' 'Proud' 'Trust' 'Sadness' 'Joy' 'Anger' 'Fear' 'Neutral']


In [3]:
import pandas as pd

# Path ke file output
output_path = '../data/processed/train_clean.csv'

try:
    # Load the cleaned CSV file
    df_clean = pd.read_csv(output_path)

    print("--- Mengecek Keunikan Kolom 'emotion' ---")
    # Hitung nilai unik dan frekuensinya di kolom 'emotion'
    emotion_unique_counts = df_clean['emotion'].value_counts()
    print("Nilai unik dan jumlahnya di kolom 'emotion':")
    print(emotion_unique_counts)

    # Periksa apakah semua nilai emosi sudah sesuai dengan daftar yang diinginkan
    print("\nNilai unik yang tersisa di kolom 'emotion' setelah normalisasi:")
    print(df_clean['emotion'].unique())

    print("\n--- Mengecek Keunikan Kolom 'id' ---")
    # Periksa apakah ada ID duplikat
    total_rows = len(df_clean)
    unique_ids = df_clean['id'].nunique()
    if total_rows == unique_ids:
        print(f"Jumlah ID unik ({unique_ids}) sama dengan jumlah baris ({total_rows}). Semua ID unik.")
    else:
        print(f"ADA ID duplikat! Jumlah ID unik ({unique_ids}) tidak sama dengan jumlah baris ({total_rows}).")

    print("\n--- Mengecek Informasi DataFrame ---")
    print("Informasi umum tentang DataFrame:")
    print(df_clean.info())
    
except FileNotFoundError:
    print(f"Error: File '{output_path}' tidak ditemukan.")
    print("Pastikan file 'train_clean.csv' sudah berhasil dibuat di folder 'data/processed' terlebih dahulu.")

--- Mengecek Keunikan Kolom 'emotion' ---
Nilai unik dan jumlahnya di kolom 'emotion':
emotion
Surprise    331
Trust       184
Proud       156
Joy          53
Anger        36
Sadness      18
Fear         16
Neutral       8
Name: count, dtype: int64

Nilai unik yang tersisa di kolom 'emotion' setelah normalisasi:
['Surprise' 'Proud' 'Trust' 'Sadness' 'Joy' 'Anger' 'Fear' 'Neutral']

--- Mengecek Keunikan Kolom 'id' ---
Jumlah ID unik (802) sama dengan jumlah baris (802). Semua ID unik.

--- Mengecek Informasi DataFrame ---
Informasi umum tentang DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 802 entries, 0 to 801
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   id       802 non-null    int64 
 1   video    802 non-null    object
 2   emotion  802 non-null    object
dtypes: int64(1), object(2)
memory usage: 18.9+ KB
None


### Catatan Dataset

* Dataset tidak seimbang:

  * Label **"Surprise"** paling banyak
  * Label **"Natural/Neutral"** paling sedikit → perlu balancing
* Kategori label emosi:

  * `['Surprise', 'Proud', 'Trust', 'Sadness', 'Joy', 'Anger', 'Fear', 'Neutral']`
* Jumlah data:

  * **Train:** 802 video
  * **Test:** 200 video


In [4]:
print(df_train['emotion'].unique())

['Surprise' 'Proud' 'Trust' 'Sadness' 'Joy' 'Anger' 'Fear' 'Neutral']
